src/documents/tasks.py

```diff
@@ -195,42 +230,74 @@ def consume_file(
+            )
+        else:
+            if file_extension == ".tif" or file_extension == ".tiff":
+                converted_tiff = convert_from_tiff_to_pdf(path)
+            if converted_tiff:
+                separators = scan_file_for_separating_barcodes(converted_tiff)
+            else:
+                separators = scan_file_for_separating_barcodes(path)
+            if separators:
+                if converted_tiff:
+                    logger.debug(
+                        f"Pages with separators found in: {str(converted_tiff)}",
+                    )
+                    document_list = separate_pages(converted_tiff, separators)
                 else:
-                    newname = None
-                save_to_dir(document, newname=newname)
-            # if we got here, the document was successfully split
-            # and can safely be deleted
-            logger.debug("Deleting file {}".format(path))
-            os.unlink(path)
-            # notify the sender, otherwise the progress bar
-            # in the UI stays stuck
-            payload = {
-                "filename": override_filename,
-                "task_id": task_id,
-                "current_progress": 100,
-                "max_progress": 100,
-                "status": "SUCCESS",
-                "message": "finished",
-            }
-            try:
-                async_to_sync(get_channel_layer().group_send)(
-                    "status_updates",
-                    {"type": "status_update", "data": payload},
-                )
-            except OSError as e:
-                logger.warning("OSError. It could be, the broker cannot be reached.")
-                logger.warning(str(e))
-            return "File successfully split"
+                    logger.debug(
+                        f"Pages with separators found in: {str(path)}",
+                    )
+                    document_list = separate_pages(path, separators)
+            if document_list:
+                for n, document in enumerate(document_list):
+                    # save to consumption dir
+                    # rename it to the original filename  with number prefix
+                    if override_filename:
+                        newname = f"{str(n)}_" + override_filename
+                    else:
+                        newname = None
+                    save_to_dir(document, newname=newname)
+                # if we got here, the document was successfully split
+                # and can safely be deleted
+                if converted_tiff:
+                    logger.debug("Deleting file {}".format(converted_tiff))
+                    os.unlink(converted_tiff)
+                logger.debug("Deleting file {}".format(path))
+                os.unlink(path)
+                # notify the sender, otherwise the progress bar
+                # in the UI stays stuck
+                payload = {
+                    "filename": override_filename,
```

Comment:
Is `override_filename` the right one here?  It could be `None`?

Reply:
I don't belive it will cause problems, it looks like we already do it this way:
https://github.com/paperless-ngx/paperless-ngx/blob/cc93616019306c6a9b308ffdb1ffbc4ecbd371e9/src/documents/consumer.py#L65